src/coreclr/tools/Common/TypeSystem/Ecma/EcmaMethod.cs

```diff
@@ -499,6 +500,25 @@ public override PInvokeMetadata GetPInvokeMethodMetadata()
             if ((ImplAttributes & MethodImplAttributes.PreserveSig) != 0)
                 attributes |= PInvokeAttributes.PreserveSig;
 
+            if (Context.Target.IsOSX && moduleName == "/usr/lib/libobjc.dylib")
```

Comment:
Can we avoid hardcoding knowledge of this into the type system (i.e. can this pull request get by without changing anything under src/coreclr/tools/Common/TypeSystem/Ecma)?

I'd move this logic to `MarshalHelpers`, around `ShouldCheckForPendingException`. We tend to avoid extending the type system outside of what the ECMA-335 spec has, unless there's perf reasons.

Reply:
I moved the logic for detecting the type of objc_msgSend function to the `MarshalHelpers` as you suggested.